node/src/components/contract_runtime.rs

```diff
@@ -202,362 +298,341 @@ where
+                            let correlation_id = CorrelationId::new();
+                            let start = Instant::now();
+                            let result = engine_state.get_bids(correlation_id, get_bids_request);
+                            metrics.get_bids.observe(start.elapsed().as_secs_f64());
+                            trace!(?result, "get bids result");
+                            responder.respond(result).await
+                        }
+                        .ignore()
+                    }
+                    ContractRuntimeRequest::MissingTrieKeys {
+                        trie_key,
+                        responder,
+                    } => {
+                        trace!(?trie_key, "missing_trie_keys request");
+                        let engine_state = Arc::clone(&self.engine_state);
+                        let metrics = Arc::clone(&self.metrics);
+                        async move {
+                            let correlation_id = CorrelationId::new();
+                            let start = Instant::now();
+                            let result =
+                                engine_state.missing_trie_keys(correlation_id, vec![trie_key]);
+                            metrics.read_trie.observe(start.elapsed().as_secs_f64());
+                            trace!(?result, "missing_trie_keys response");
+                            responder.respond(result).await
+                        }
+                        .ignore()
+                    }
                 }
-                .ignore()
             }
-            Event::Request(ContractRuntimeRequest::GetBids {
-                get_bids_request,
-                responder,
-            }) => {
-                trace!(?get_bids_request, "get bids request");
-                let engine_state = Arc::clone(&self.engine_state);
-                let metrics = Arc::clone(&self.metrics);
-                async move {
-                    let correlation_id = CorrelationId::new();
-                    let result = task::spawn_blocking(move || {
-                        let start = Instant::now();
-                        let result = engine_state.get_bids(correlation_id, get_bids_request);
-                        metrics.get_bids.observe(start.elapsed().as_secs_f64());
-                        result
-                    })
-                    .await
-                    .expect("should run");
-                    trace!(?result, "get bids result");
-                    responder.respond(result).await
-                }
-                .ignore()
+            Event::BlockAlreadyExists(block) => effect_builder
+                .announce_block_already_executed(*block)
+                .ignore(),
+            // If we haven't executed the block before in the past (for example during
+            // joining), do it now.
+            Event::BlockIsNew(finalized_block) => {
+                self.get_deploys(effect_builder, *finalized_block)
```

Comment:
We are in the `async fn` so we should be able to `async move { self.get_deploys.await }` and not have to send event to ourselves, right?